account_invoice_inter_company/models/account_move.py

```diff
@@ -115,7 +117,19 @@ def _inter_company_create_invoice(self, dest_company):
         if dest_company.invoice_auto_validation and not float_compare(
             self.amount_total, dest_invoice.amount_total, precision_digits=precision
         ):
+            pdf = self.env.ref("account.account_invoices")._render_qweb_pdf(self.id)[0]
```

Comment:
In fact it's not correct.
Indeed in case of manual validation the dest company will not have any pdf.
It's will better to inherit the method "action_post" and in if the field "auto_invoice_id" is fill, in that case call a method "_attach_original_pdf_report" that add the pdf in teh attachment.

Also having the logic in a specific method give us the possibility to use it in specific migration script

Reply:
I m not sure it is that simple...

Reply:
As discussed together, I bring another solution.